renku/core/metadata/database.py

```diff
@@ -80,9 +86,25 @@ def get_class(type_name: Optional[str]) -> Optional[type]:
 
 def get_attribute(object, name: Union[List[str], str]):
     """Return an attribute of an object."""
+    import sys
+
     components = name.split(".") if isinstance(name, str) else name
 
+    def _module_name(o):
+        return o.__module__ if hasattr(o, "__module__") else o.__name__
+
+    module_name = _module_name(object)
+    root_module_name = module_name.split(".")[0]
+
     for component in components:
+        module_name = _module_name(object)
+        if not hasattr(object, component) and f"{module_name}.{component}" not in sys.modules:
+            try:
+                _is_module_allowed(root_module_name, object.__name__)
+                __import__(f"{module_name}.{component}")
```

Comment:
Looking at https://docs.python.org/3/library/importlib.html#importlib.__import__ it says `Programmatic importing of modules should use import_module() instead of this function.`

The main difference is that `importlib.import_module` returns the imported thing, so we can use it directly and save the `getattr()` call below